exonum-java-binding-core/src/test/java/com/exonum/binding/storage/indices/ProofMapIndexProxyIntegrationTest.java

```diff
@@ -234,37 +230,33 @@ public void getRootHash_NonEmptyMap() {
   }
 
   @Test
-  @Ignore
   public void getProof_EmptyMap() {
     runTestWithView(database::createSnapshot,
-        (map) -> assertThat(map, provesNoMappingFor(PK1))
+        (map) -> assertThat(map, provesNoMappingFor(PK1, map.getRootHash()))
```

Comment:
As our matcher receives the map as its argument and can get the root hash in its `matchesSafely` method (as it did), why the callers are required to do `map.getRootHash()` each time they need to use this matcher?

Reply:
I don't like this approach as well and I'm still wondering what is the better way to do it.
Indeed we can get root hash in `matchesSafely` method, but how do we proceed from there? We can't pass it to `CheckedMapProofMatcher` (unless we make some sort of pair class `<CheckedMapProof, HashCode>` so it could be used as a generic in `CheckedMapProofMatcher`). Do we create separate matcher for `HashCode`? Or just
```
  @Override
  protected boolean matchesSafely(ProofMapIndexProxy<HashCode, String> map) {
    CheckedMapProof checkedProof = checkProof(map);
    HashCode expectedRootHash = map.getRootHash();

    return checkedMapProofMatcher.matches(checkedProof) && checkedMapProof.compareWithRootHash(expectedRootHash);
  }
```